snapcraft/extensions/registry.py

```diff
@@ -16,15 +16,19 @@
 
 """Extension registry."""
 
-from typing import Dict, List, Type
+from typing import TYPE_CHECKING, Dict, List, Type
 
 from snapcraft import errors
 
-from ._extension import Extension
+from .gnome import GNOME
 
-ExtensionType = Type[Extension]
+if TYPE_CHECKING:
+    from .extension import Extension
 
-_EXTENSIONS: Dict[str, ExtensionType] = {}
+    ExtensionType = Type[Extension]
+
+
+_EXTENSIONS: Dict[str, "ExtensionType"] = {"gnome": GNOME}
```

Comment:
Does this mean we won't have a way to use an extension to use a future platform snap, like gnome-44-2204 that is still based on core22?

Reply:
this is what we agreed to, remember? If it came to it, it would be a transparent upgrade for users

Reply:
I know, and I like the idea in principle but it doesn't give us an easy way to build things that need something newer versions before the next LTS.  Could we add an optional flag of some sort that let's us override the version of the content and build snaps?

Reply:
latest commit has a mechanism